src/app/common/TrayView.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div
+    class="tray"
+    data-testid="tray"
+  >
+    <header class="tray-header">
+      <div class="tray-title-wrapper">
+        <span
+          v-if="$slots.icon"
+          class="tray-icon"
+        >
+          <slot name="icon" />
+        </span>
+
+        <h2 class="tray-title">
+          <slot name="title" />
+        </h2>
+      </div>
+
+      <KButton
+        appearance="btn-link"
+        class="non-visual-button"
+        data-testid="close-tray-button"
+        @click="emit('close')"
+      >
+        <CloseIcon aria-hidden="true" />
+
+        <span class="visually-hidden">{{ t('common.tray.close') }}</span>
+      </KButton>
+    </header>
+
+    <slot />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { CloseIcon } from '@kong/icons'
+
+import { useI18n } from '@/utilities'
+
+const { t } = useI18n()
+
+const emit = defineEmits<{
+  (event: 'close'): void
+}>()
+</script>
+
+<style lang="scss" scoped>
+.tray {
+  position: fixed;
+  z-index: 100;
+  top: var(--AppHeaderHeight);
+  bottom: 0;
+  right: 0;
+  width: 600px;
+  max-width: calc(100% - var(--AppSidebarWidth));
+  padding: $kui-space-70;
+  border-left: $kui-border-width-10 solid $kui-color-border;
+  background-color: $kui-color-background;
+  box-shadow: -6px 0px 15px 0px rgba(0, 0, 0, 0.04);
+}
+
+.tray-header {
+  display: flex;
+  align-items: center;
+  gap: $kui-space-50;
+  justify-content: space-between;
+  margin-bottom: $kui-space-50;
+}
+
+.tray-title-wrapper {
+  display: flex;
+  align-items: center;
+  gap: $kui-space-30;
+}
+
+.tray-title {
+  margin-top: 0;
+}
+</style>
```

Comment:
I found a few issues with it. Some aren’t immediately fixable with some tactical overrides:

- Has no `title` slot, only a string prop. This way, we can’t use a link in the title (I want to link to the detail view here because the tray will always hide the table’s context menu while it’s open).
- `props.offsetTop` only accepts a number but we want to pass the string`var(--AppHeaderHeight)`. **Workaround**: Override with CSS.
- The component will always trigger a close action when clicking outside of it. That’s inconvenient for us because we would like to let the user select another row without first closing the tray.
- We can add an icon using the before title slot; however, its styles are awkward (no margin the left). **Workaround**: Add it with CSS.
- Uses a KCard with custom `--KCardPaddingX` and `--KCardPaddingY` values that don’t match the KCard default. **Workaround**: Override with CSS.

I added a commit with the KSlideout variant instead of comparison.

Some of the things above would be easy to supply as features or fixes in Kongponents, but those would likely only become available with Kongponents v9.

Reply:
I’m sticking to KSlideout and filed PRs for all issues mentioned above (I updated https://github.com/kumahq/kuma-gui/pull/1613#discussion_r1360627713 to list which PRs were filed for which problem).

> Then lastly for the purposes of this PR, do you think we need to wrap the kongponent in an extra wrapper component?

Absolutely. I don’t want to leave the matter of which exact props to set how to each implementation. We can abstract this problem away so I will.

Reply:
Ok sounds good, just getting in but let me take another look over this now taking this conversation into account.